pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -1074,17 +1074,13 @@ func setTimeout(action *route.RouteAction, vsTimeout *duration.Duration, node *m
 			MaxStreamDuration: action.Timeout,
 		}
 	} else {
-		// Set MaxStreamDuration only for notimeout cases otherwise it wont be honored.
+		// If not configured at all, the grpc-timeout header is not used and
+		// gRPC requests time out like any other requests using timeout or its default.
+		// Use deprecated value for now as the replacement MaxStreamDuration has some regressions.
+		// nolint: staticcheck
 		if action.Timeout.AsDuration().Nanoseconds() == 0 {
-			action.MaxStreamDuration = &route.RouteAction_MaxStreamDuration{
```

Comment:
thanks!
this is what I meant by https://github.com/istio/istio/pull/42035#discussion_r1025076719

Reply:
Thanks a lot @ramaraochavali. We have been hit with this issue as well.